paws_collector.js

```diff
@@ -890,8 +897,77 @@ class PawsCollector extends AlAwsCollector {
             return asyncCallback(err);
         });
     }
+    /**
+     * Return the original error after deleting the entry from DDB
+     * @param {*} error 
+     * @param {*} logs 
+     * @param {*} asyncCallback 
+     * @returns 
+     */
+    handleDeDupIngestError(error, logs, asyncCallback) {
+        let collector = this;
+        if (collector.pawsCollectorType === 'o365') {
+            collector.deleteDedupLogItemEntry(logs, (err) => {
+                if (err) {
+                    AlLogger.warn(`PAWS000406 Error while delete item in DynamoDB ${err}`);
+                }
+                return asyncCallback(error);
+            });
+        } else {
+            return asyncCallback(error);
+        }
+    }
+    /**
+     * delete the item entry from DDB
+     * @param {*} logs 
+     * @param {*} callback 
+     */
+    deleteDedupLogItemEntry(logs, callback) {
+        let collector = this;
+        const cid = collector.cid ? collector.cid : 'none';
+        const collectorId = collector._collectorId;
+        const tableName = collector._pawsDeDupLogsTableName
+        const promises = [];
+        let currentBatch = [];
+
+        for (const message of logs) {
+            const messageHash = collector.getHash(message);
+            const itemId = `${cid}_${collectorId}_${messageHash}`;
```

Comment:
let's have a separate function to make the item ID so we don't replicate logic between here and `removeDuplicatedItem()`